Sanitizer error in ydb/tests/tools/nemesis/ut/test_tablet.py.TestMassiveKills.test_tablets_are_ok_after_many_kills

ydb/tests/tools/nemesis/ut/test_tablet.py.TestMassiveKills.test_tablets_are_ok_after_many_kills

**Build Information:**
- Version: ef902594170e0598519a2365c6e903cad100386a
- Build Preset: release-asan
- Branch: stable-25-3-1

**Run Command:**
```
./ya make -ttt --build "release" --sanitize="address" -DDEBUGINFO_LINES_ONLY -F 'test_tablet.py::TestMassiveKills::test_tablets_are_ok_after_many_kills' ydb/tests/tools/nemesis/ut
```

**Owner:** [TEAM:@ydb-platform/](https://github.com/orgs/ydb-platform/teams/)

**Test run history:** [link](https://datalens.yandex/34xnbsom67hcq?branch=stable-25-3-1&full_name=ydb/tests/tools/nemesis/ut/test_tablet.py.TestMassiveKills.test_tablets_are_ok_after_many_kills&build_type=release-asan)

More info in [dashboard](https://datalens.yandex/4un3zdm0zcnyr)

**Logs:**
- [log](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_Small_and_Medium/17944792087/ya-stable-25-3-1-x86-64-asan/try_2/artifacts/logs/ydb/tests/tools/nemesis/ut/test-results/py3test/testing_out_stuff/test_tablet.py.TestMassiveKills.test_tablets_are_ok_after_many_kills.log)
- [logsdir](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_Small_and_Medium/17944792087/ya-stable-25-3-1-x86-64-asan/try_2/test_artifacts/ydb-tests-tools-nemesis-ut.zip)

**Error Summary:**
```
E   ydb.tests.library.harness.daemon.SeveralDaemonErrors: Daemon failed with message: Unexpectedly finished before stop.
E   Process exit_code = 100.
E   Stdout file name: 
E   /home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/tools/nemesis/ut/test-results/py3test/testing_out_stuff/test_tablet.py.TestMassiveKills.test_tablets_are_ok_after_many_kills/cluster/node_7/stdout
E   Stderr file name: 
E   /home/runner/actions_runner/_work/ydb/ydb/tmp/out/ydb/tests/tools/nemesis/ut/test-results/py3test/testing_out_stuff/test_tablet.py.TestMassiveKills.test_tablets_are_ok_after_many_kills/cluster/node_7/stderr
E   Stderr content:
E   
E   GRpc memory quota was set but disabled due to issues with grpc quoter, to enable it use EnableGRpcMemoryQuota option
E   =================================================================
E   ==1722671==ERROR: AddressSanitizer: heap-use-after-free on address 0x7b9838c34049 at pc 0x00001f6afd83 bp 0x7b67ed31e710 sp 0x7b67ed31ded0
E   READ of size 2 at 0x7b9838c34049 thread T51 (ydbd.System)
E   warning: address range table at offset 0x10c0 has a premature terminator entry at offset 0x10d0
E   #0 0x00001f6afd82 in __asan_memmove /-S/contrib/libs/clang20-rt/lib/asan/asan_interceptors_memintrinsics.cpp:71:3
E   #1 0x00001f87b764 in __constexpr_memmove<char, const char, 0> /-S/contrib/libs/cxxsupp/libcxx/include/__string/constexpr_c_functions.h:229:5
E   #2 0x00001f87b764 in copy /-S/contrib/libs/cxxsupp/libcxx/include/__string/char_traits.h:249:5
E   #3 0x00001f87b764 in std::__y1::basic_string<char, std::__y1::char_traits<char>, std::__y1::allocator<char>>::append(char const*, unsigned long) /-S/contrib/libs/cxxsupp/libcxx/include/string:2879:7
E   #4 0x00004cbb68e7 in operator<<<TBasicStringBuf<char, std::__y1::char_traits<char> > > /-S/util/stream/output.h:238:5
E   #5 0x00004cbb68e7 in operator<<<TBasicStringBuf<char, std::__y1::char_traits<char> > > /-S/util/string/builder.h:26:21
E   #6 0x00004cbb68e7 in NKikimr::NHive::THiveBalancer::Handle(TAutoPtr<NActors::TEventHandle<NKikimr::NHive::TEvPrivate::TEvRestartComplete>, TDelete>&) /-S/ydb/core/mind/hive/balancer.cpp:345:9
E   #7 0x00002219117b in NActors::IActor::Receive(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/core/actor.cpp:406:17
E   #8 0x000022248e21 in NActors::TExecutorThread::Execute(NActors::TMailbox*, bool) /-S/ydb/library/actors/core/executor_thread.cpp:268:28
E   #9 0x000022252a14 in NActors::TExecutorThread::ProcessExecutorPool()::$_0::operator()(NActors::TMailbox*, bool) const /-S/ydb/library/actors/core/executor_thread.cpp:458:39
E   #10 0x000022251fcd in NActors::TExecutorThread::ProcessExecutorPool() /-S/ydb/library/actors/core/executor_thread.cpp:510:13
E   #11 0x00002225403e in NActors::TExecutorThread::ThreadProc() /-S/ydb/library/actors/core/executor_thread.cpp:536:9
E   #12 0x00001fa11ba4 in (anonymous namespace)::TPosixThread::ThreadProxy(void*) /-S/util/system/thread.cpp:245:20
E   #13 0x00001f6af466 in asan_thread_start(void*) /-S/contrib/libs/clang20-rt/lib/asan/asan_interceptors.cpp:239:28
E   #14 0x7f6833a29ac2  (/lib/x86_64-linux-gnu/libc.so.6+0x94ac2) (BuildId: d5197096f709801829b118af1b7cf6631efa2dcd)
E   #15 0x7f6833abb84f  (/lib/x86_64-linux-gnu/libc.so.6+0x12684f) (BuildId: d5197096f709801829b118af1b7cf6631efa2dcd)
E   
E   0x7b9838c34049 is located 9 bytes inside of 32-byte region [0x7b9838c34040,0x7b9838c34060)
E   freed by thread T51 (ydbd.System) here:
E   #0 0x00001f6ec5b2 in operator delete(void*, unsigned long) /-S/contrib/libs/clang20-rt/lib/asan/asan_new_delete.cpp:155:3
E   #1 0x00004cbf97c2 in UnRef /-S/util/generic/string.h:105:13
E   #2 0x00004cbf97c2 in UnRef /-S/util/generic/string.h:58:16
E   #3 0x00004cbf97c2 in UnRef /-S/util/generic/ptr.h:641:13
E   #4 0x00004cbf97c2 in ~TIntrusivePtr /-S/util/generic/ptr.h:539:9
E   #5 0x00004cbf97c2 in ~TBasicString /-S/util/generic/fwd.h:8:7
E   #6 0x00004cbf97c2 in NKikimr::NHive::TTxUpdateTabletStatus::Execute(NKikimr::NTabletFlatExecutor::TTransactionContext&, NActors::TActorContext const&) /-S/ydb/core/mind/hive/tx__update_tablet_status.cpp:100:17
E   #7 0x0000272986ce in NKikimr::NTabletFlatExecutor::TExecutor::ExecuteTransaction(NKikimr::NTabletFlatExecutor::TSeat*) /-S/ydb/core/tablet_flat/flat_executor.cpp:2026:35
E   #8 0x0000272929ea in NKikimr::NTabletFlatExecutor::TExecutor::DoExecute(TAutoPtr<NKikimr::NTabletFlatExecutor::ITransaction, TDelete>, NKikimr::NTabletFlatExecutor::TExecutor::ETxMode) /-S/ydb/core/tablet_flat/flat_executor.cpp:1931:13
E   #9 0x00002729b2fe in NKikimr::NTabletFlatExecutor::TExecutor::Execute(TAutoPtr<NKikimr::NTabletFlatExecutor::ITransaction, TDelete>, NActors::TActorContext const&) /-S/ydb/core/tablet_flat/flat_executor.cpp:1945:5
E   #10 0x000027230783 in NKikimr::NTabletFlatExecutor::TTabletExecutedFlat::Execute(TAutoPtr<NKikimr::NTabletFlatExecutor::ITransaction, TDelete>) /-S/ydb/core/tablet_flat/tablet_flat_executed.cpp:62:46
E   #11 0x00004c79fd08 in NKikimr::NHive::THive::Handle(TAutoPtr<NActors::TEventHandle<NKikimr::TEvLocal::TEvTabletStatus>, TDelete>&) /-S/ydb/core/mind/hive/hive_impl.cpp:507:9
E   #12 0x00004c7ffa67 in NKikimr::NHive::THive::ProcessEvent(std::__y1::unique_ptr<NActors::IEventHandle, std::__y1::default_delete<NActors::IEventHandle>>) /-S/ydb/core/mind/hive/hive_impl.cpp:3163:9
E   #13 0x00004c7fecdf in NKikimr::TEventPriorityQueue<NKikimr::NHive::THive>::ProcessIncomingEvent() /-S/ydb/core/util/event_priority_queue.h:67:16
E   #14 0x00004c8148f8 in Handle /-S/ydb/core/mind/hive/hive_impl.cpp:3101:16
E   #15 0x00004c8148f8 in NKikimr::NHive::THive::StateWork(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/core/mind/hive/hive_impl.cpp:3357:9
E   #16 0x00002219117b in NActors::IActor::Receive(TAutoPtr<NActors::IEventHandle, TDelete>&) /-S/ydb/library/actors/core/actor.cpp:406:17
E   #17 0x000022248e21 in NActors::TExecutorThread::Execute(NActors::TMailbox*, bool) /-S/ydb/library/actors/core/executor_thread.cpp:268:28
E   #18 0x000022252a14 in NActors::TExecutorThread::ProcessExecutorPool()::$_0::operator()(NActors::TMailbox*, bool) const /-S/ydb/library/actors/core/executor_thread.cpp:458:39
E   #19 0x000022251fcd in NActors::TExecutorThread::ProcessExecutorPool() /-S/ydb/library/actors/core/executor_thread.cpp:510:13
E   #20 0x00002225403e in NActors::TExecutorThread::ThreadProc() /-S/ydb/library/actors/core/executor_thread.cpp:536:9
E   #21 0x00001fa11ba4 in (anonymous namespace)::TPosixThread::ThreadProxy(void*) /-S/util/system/thread.cpp:245:20
E   #22 0x00001f6af466 in asan_thread_start(void*) /-S/contrib/libs/clang20-rt/lib/asan/asan_interceptors.cpp:239:28
E   
E   previously allocated by thread T51 (ydbd.System) here:
E   #0 0x00001f6eb94d in operator new(unsigned long) /-S/contrib/libs/clang20-rt/lib/asan/asan_new_delete.cpp:86:3
E   #1 0x00004cbf9655 in Construct<const char *&, unsigned long &> /-S/util/generic/string.h:212:17
E   #2 0x00004cbf9655 in TBasicString /-S/util/generic/string.h:491:18
E   #3 0x00004cbf9655 in TBasicString /-S/util/generic/string.h:482:11
E   #4 0x00004cbf9655 in NKikimr::NHive::TTxUpdateTabletStatus::Execute(NKikimr::NTabletFlatExecutor::TTransactionContext&, NActors::TActorContext const&) /-S/ydb/core/mind/hive/tx__update_tablet_status.cpp:100:41
E   #5 0x0000272986ce in NKikimr::NTabletFlatExecutor::TExecutor::ExecuteTransaction(NKikimr::NTabletFlatExecutor::TSeat*) /-S/ydb/core/tablet_flat/flat_executor.cpp:2026:35
E   #6 0x0000272929ea in NKikimr::NTabletFlatExecutor::TExecutor::DoExecute(TAutoPtr<NKikimr::NTabletFlatExecutor::ITransaction, TDelete>, NKikimr::NTabletFlatExecutor::TExecutor::ETxMode) /-S/ydb/core/tablet_flat/flat_executor.cpp:1931:13
E   #7 0x00002729b2fe in NKikimr::NTabletFlatExecutor::TExecutor::Execute(TAutoPtr<NKikimr::NTabletF...
```

--- a/ydb/core/mind/hive/hive_events.h
+++ b/ydb/core/mind/hive/hive_events.h
@@ -89,9 +89,9 @@ struct TEvPrivate {
 
     struct TEvRestartComplete : TEventLocal<TEvRestartComplete, EvRestartComplete> {
         TFullTabletId TabletId;
-        TStringBuf Status;
+        TString Status;
 
-        TEvRestartComplete(TFullTabletId tabletId, TStringBuf status)
+        TEvRestartComplete(TFullTabletId tabletId, const TString& status)
             : TabletId(tabletId)
             , Status(status)
         {}

--- a/ydb/core/mind/hive/tablet_info.cpp
+++ b/ydb/core/mind/hive/tablet_info.cpp
@@ -545,7 +545,7 @@ bool TTabletInfo::RestartsOften() const {
     return Statistics.RestartTimestampSize() >= Hive.GetTabletRestartsMaxCount();
 }
 
-void TTabletInfo::NotifyOnRestart(TString status, TSideEffects& sideEffects) {
+void TTabletInfo::NotifyOnRestart(const TString& status, TSideEffects& sideEffects) {
     for (auto actorId : ActorsToNotifyOnRestart) {
         sideEffects.Send(actorId, new TEvPrivate::TEvRestartComplete(GetFullTabletId(), status));
     }

--- a/ydb/core/mind/hive/tablet_info.h
+++ b/ydb/core/mind/hive/tablet_info.h
@@ -310,7 +310,7 @@ public:
         return std::get<NMetrics::EResource::Counter>(GetResourceCurrentValues()) > 0;
     }
 
-    void NotifyOnRestart(TString status, TSideEffects& sideEffects);
+    void NotifyOnRestart(const TString& status, TSideEffects& sideEffects);
 };
 
 

